beetmoverscript/script.py

```diff
@@ -33,26 +35,38 @@
     # balrogworker task in the release graph. Balrogworker uses this manifest to submit
     # release blob info with things like mar filename, size, etc
     context.balrog_manifest = list()
+    # checksums_manifest is written and uploaded as an artifact which is used
+    # by a subsequent signing task and again by another beetmover task to
+    # upload it along with the other artifacts
+    context.checksums_manifest = list()
     # determine and validate the task schema along with its scopes
     context.task = get_task(context.config)  # e.g. $cfg['work_dir']/task.json
     validate_task_schema(context)
     # determine artifacts to beetmove
     context.artifacts_to_beetmove = get_upstream_artifacts(context)
     # determine the release properties
     context.release_props = get_release_props(get_initial_release_props_file(context))
+    # determine the variables to feed the manifest templates
+    template_args = generate_beetmover_template_args(context.task,
+                                                     context.release_props)
     # generate beetmover mapping manifest
     mapping_manifest = generate_beetmover_manifest(context.config,
-                                                   context.task,
-                                                   context.release_props)
+                                                   template_args)
     # validate scopes to prevent beetmoving in the wrong place
     validate_task_scopes(context, mapping_manifest)
-    # 5. for each artifact in manifest
+    # for each artifact in manifest
     #   a. map each upstream artifact to pretty name release bucket format
     #   b. upload to candidates/dated location
     await move_beets(context, context.artifacts_to_beetmove, mapping_manifest)
-    # # 6. write balrog_manifest to a file and add it to list of artifacts
+    #  write balrog_manifest to a file and add it to list of artifacts
     if context.task["payload"]["update_manifest"]:
         add_balrog_manifest_to_artifacts(context)
+    # determine the correct checksum filename and generate it, adding it to
+    # the list of artifacts afterwards
+    if context.task["payload"]["generate_checksums"]:
```

Comment:
aki mentioned that we may not need this boolean here: https://bugzilla.mozilla.org/show_bug.cgi?id=1305139#c7